runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java

```diff
@@ -301,7 +303,17 @@ private void updateShownIcons()
 				}
 			}).map(TeleportPoint::new)
 			.forEach(worldMapPointManager::add);
-	}
+
+			worldMapPointManager.removeIf(RunecraftingAltarPoint.class::isInstance);
+
+			if (config.runecraftingAltarIcon())
+			{
+				Arrays.stream(RunecraftingAltarLocation.values())
+					.filter(altarLocation -> altarLocation.getLocation() != null)
```

Comment:
Will there ever be altars with null locations? There aren't currently any like that in the added enum.